File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>
#include <map>


using namespace std;
char s[8] = {'R', 'T', 'C', 'F', 'J', 'M', 'A', 'N'};

string solution(vector<string> survey, vector<int> choices) {
        map<char, int> m;
    for (int i = 0; i < 8; ++i) {
        m.insert({s[i], 0});
    }
    for (int i = 0; i < survey.size(); ++i) {
        int num = choices[i];
        char f = survey[i][0];
        char r = survey[i][1];
        if (num < 4) {
            m[f] += 4 - num;
        } else if (num > 4) {
            m[r] += num - 4;
        }
    }
    string answer = "";
        for (int i = 0; i < 4; i++) {
        int a = 2 * i, b = a + 1;
        if(m[s[a]] > m[s[b]])
            answer += s[a];
        else if(m[s[a]] < m[s[b]])
            answer += s[b];
        else
            answer += s[a] > s[b] ? s[b] : s[a];
    }
    
    return answer;
    
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        int len = survey.length;
        Map<String, Integer> map = new HashMap<>();
        String[][] arr = {{"R", "T"}, {"C", "F"}, {"J", "M"}, {"A", "N"}};
        
        
        
        for(String[] a : arr) {
            for(String ch : a ) {
                map.put(ch, 0);            
            }
        }
        
        
    
        for(int i = 0; i < len; i++ ) {
            String sur = survey[i];
            String f = sur.substring(0, 1);
            String b = sur.substring(1, 2);
            int choice = choices[i];
            
            if(choice < 4) {
                map.put(f, map.get(f) + 4 - choice);
                continue;
            }
            
            if(choice > 4) {
                map.put(b, map.get(b) + choice - 4);
            }
            
            
        }
        
        StringBuilder sb = new StringBuilder();
        for(String[] type : arr) {
            int fscore = map.get(type[0]);
            int bscore = map.get(type[1]);
            
            if(fscore >= bscore) {
                sb.append(type[0]);
            } else {
                sb.append(type[1]);
            }
            
        }
        
        return sb.toString();
    }
}